checker/src/org/checkerframework/checker/samelen/SameLenAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,268 @@
+                }
+            }
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2) {
+            if (AnnotationUtils.hasElementValue(a1, "value")
+                    && AnnotationUtils.hasElementValue(a2, "value")) {
+                String[] a1Val = SameLenUtils.getValue(a1);
+                String[] a2Val = SameLenUtils.getValue(a2);
+
+                if (overlap(a1Val, a2Val)) {
+                    return getCombinedSameLen(a1Val, a2Val);
+                } else {
+                    return createSameLenUnknown();
+                }
+
+            } else {
+                // the lub is either one of the annotations (if the other is bottom), or top.
+                if (AnnotationUtils.areSameByClass(a1, SameLenBottom.class)) {
+                    return a2;
+                } else if (AnnotationUtils.areSameByClass(a2, SameLenBottom.class)) {
+                    return a1;
+                } else {
+                    return createSameLenUnknown();
+                }
+            }
+        }
+
+        /**
+         * Computes subtyping. First checks if one is bottom or the other top (since top is a
+         * subtype of nothing, but everything is a subtype of it, and bottom is a subtype of
+         * everything, but nothing is a subtype of it. Then, checks if the types are the same. If
+         * they are, return true. Otherwise, they're distinct, so return false.
+         *
+         * @return true if rhs is a subtype of lhs, false otherwise
+         */
+        @Override
+        public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
+            if (AnnotationUtils.areSameByClass(rhs, SameLenBottom.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(lhs, SameLenUnknown.class)) {
+                return true;
+            }
+            if (AnnotationUtils.hasElementValue(rhs, "value")
+                    && AnnotationUtils.hasElementValue(lhs, "value")) {
+                String[] a1Val = SameLenUtils.getValue(rhs);
+                String[] a2Val = SameLenUtils.getValue(lhs);
+
+                if (overlap(a1Val, a2Val)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
+
+    @Override
+    public TreeAnnotator createTreeAnnotator() {
+        return new ListTreeAnnotator(
+                new SameLenTreeAnnotator(this),
+                new PropagationTreeAnnotator(this),
+                new ImplicitsTreeAnnotator(this));
+    }
+
+    protected class SameLenTreeAnnotator extends TreeAnnotator {
+        public SameLenTreeAnnotator(SameLenAnnotatedTypeFactory factory) {
+            super(factory);
+        }
+    }
+
+    public AnnotationMirror createSameLen(String val) {
+        String[] vals = {val};
+        return createSameLen(vals);
+    }
+
+    public AnnotationMirror createSameLen(String[] val) {
```

Comment:
Use a varargs so that you don't need both createSameLen methods. 

`createSameLen(String... val)`